core/__tests__/models/model/modelCache.ts

```diff
@@ -0,0 +1,129 @@
+import { helper } from "@grouparoo/spec-helper";
+import { GrouparooModel } from "../../../src";
+import { ModelsCache } from "../../../src/modules/caches/modelsCache";
+
+describe("models/modelsCache", () => {
+  helper.grouparooTestServer({ truncate: true, enableTestPlugin: true });
+
+  beforeEach(() => (ModelsCache.TTL = 30 * 1000));
+  afterAll(() => (ModelsCache.expires = 0));
+
+  beforeAll(async () => {
+    const deletedModel = await helper.factories.model();
+    await deletedModel.update({ state: "deleted" });
+  });
+
+  describe("#findAllWithCache", () => {
+    test("it returns all the instances", async () => {
+      const instances = await ModelsCache.findAllWithCache();
+      expect(instances.length).toBe(1);
+    });
+
+    test("it can filter to state", async () => {
+      const newModel = await helper.factories.model({ name: "foo" });
+      await newModel.update({ state: "ready" }, { hooks: false });
+
+      const instances = await ModelsCache.findAllWithCache(undefined, "ready");
+
+      expect(instances.length).toBe(1);
+      expect(instances[0].id).toBe(newModel.id);
+
+      await newModel.destroy();
+    });
+
+    describe("rpc", () => {
+      let model: GrouparooModel;
+
+      async function makeModel() {
+        model = await helper.factories.model({ name: "foo" });
+        await helper.sleep(100); // wait for delayed RPC calls
+      }
+
+      afterEach(async () => {
+        await model.destroy().catch(() => null);
+      });
+
+      test("creating a model signals RPC", async () => {
+        ModelsCache.expires = new Date().getTime();
+        await makeModel();
+        await helper.sleep(10);
+        expect(ModelsCache.expires).toBe(0);
+      });
```

Comment:
Are these actually testing the RPC signaling? Seems like the `XYZ.invalidateCache()` method invalidates the cache locally and then triggers the RPC `afterCommit`. (so `ModelsCache.expires` would be cleared anyway whether rpc works or not)

For app disconnection we were _only_ signaling the RPC and letting the local disconnection happen through that as well, but in this case since you want to delay invalidating the rest `afterCommit`, implementation seems ok. This is probably a rename of these tests to something like "creating a model invalidates the cache" and add a test for the rpc call itself?

Reply:
Good feedback! I've added better RPC tests via 702841e55.